examples/gno.land/p/demo/uuid/uuid.gno

```diff
@@ -0,0 +1,125 @@
+// distributed systems. The package offers a mechanism to convert these IDs
+// to standard UUID format strings, making them compatible with systems
+// requiring UUIDs.
+// Add import "gno.land/p/demo/entropy" to use the entropy package. For of the pseudo-random number generator.
+// TimesTampLength is 32 because entropy.Value() returns a uint32.
+// 32 it's the maximum value that can be returned "4 294 967 295", It's represent 136 years (4 294 967 295 secondes)
+
+package uuid
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"std"
+
+	"gno.land/p/demo/entropy"
+	"gno.land/p/demo/ufmt"
+)
+
+const (
+	TimestampLength uint8  = 32
+	MachineIDLength uint64 = 10
+	SequenceLength  uint64 = 12
+	MaxSequence     uint16 = 1<<SequenceLength - 1
+	MaxTimestamp    uint32 = 1<<TimestampLength - 1
+	MaxMachineID    uint64 = 1<<MachineIDLength - 1
+
+	machineIDMoveLength = SequenceLength
+	timestampMoveLength = MachineIDLength + SequenceLength
+
+	startTime uint32 = 1226354400 // 10 Nov 2008 23:00:00 UTC in seconds (Snowflake epoch)
+)
+
+// UUID struct for generating unique identifiers.
+type UUID struct {
+	machineID     uint64
+	sequence      uint16
+	startTime     uint32
+	lastTimestamp uint32
+	counter       int64
+	entropy       *entropy.Instance
+}
+
+func NewUUID() *UUID {
+	u := &UUID{
+		startTime: startTime,
+		entropy:   entropy.New(),
+	}
+	u.SetMachineID()
+	return u
+}
+
+// ID generates
+func (u *UUID) ID() uint64 {
+	current := u.getEntropy()
+	elapsedTime := current - u.startTime
+	u.lastTimestamp = current
+
+	if elapsedTime < 0 || elapsedTime > MaxTimestamp {
+		panic("Exceeding the time limit")
+	}
+
+	id := (uint64(elapsedTime) << timestampMoveLength) |
+		(uint64(u.machineID) << machineIDMoveLength) |
+		uint64(u.incrementSequence())
+	return id
+}
+
+func (u *UUID) incrementSequence() uint16 {
+	u.sequence = (u.sequence + 1) & MaxSequence
+	return u.sequence
+}
+
+func (u *UUID) getEntropy() uint32 {
+	return u.entropy.Value()
+}
+
+// SetMachineID sets the machine ID based on the caller’s address.
+func (u *UUID) SetMachineID() {
+	caller := std.PrevRealm().Addr() // Retrieve the caller’s address
+	machineID := uint64(0)
+	for _, c := range caller.String() {
+		machineID += uint64(c)
+	}
+	machineID %= MaxMachineID
+	u.machineID = machineID
+}
+
+func IDToUUID(id uint64) string {
+	bytes := make([]byte, 16)
+
+	// Copy transformed ID into the second half of the array
+	copy(bytes[8:], uint64ToBytes(id))
+
+	// Use bits from the Snowflake ID to generate the first half of the UUID
+	bytes[0] = byte(id >> 60)
+	bytes[1] = byte(id >> 52)
+	bytes[2] = byte(id >> 44)
+	bytes[3] = byte(id >> 36)
+	bytes[4] = byte(id >> 28)
+	bytes[5] = byte(id >> 20)
+	bytes[6] = byte(id >> 12)
+	bytes[7] = byte(id >> 4)
+
+	// set the version and variant bits according to UUID specification.
+	bytes[6] = (bytes[6] & 0x0f) | 0x40 // version 4 (random)
+	bytes[8] = (bytes[8] & 0x3f) | 0x80 // variant 1 (RFC 4122)
```

Comment:
Note, we're using the same `id` both in `uint64ToBytes`, and then copying it over in a slightly modified manner later in this function

I don't particularly like this. We're restricting the number of bits allotted to a timestamp to 32 bits, so we can generate 64-bit integers; when really UUIDs are 122-bits, so we could reasonably have timestamps of at the very least 40 bits (which IMO is more reasonable).

Additionally, this means that there are not 2^122 possible IDs out of this package, but only 2^64.

There needs to be a decision: is this a Snowflake ID generator or is this a UUID generator? Either is fine, but understanding the target is important; and if we use UUIDs we should try to use all 122 bits at our disposal to try to create really unique identifiers.

Reply:
@DIGIX666 - as @thehowl said, why not call it `snowflakeID` then?

Reply:
Because I thought keeping the name `UUID` was more explicit, but in light of your feedback and my desire to stick with a `Snowflake ID` generator system, I will change the names to maintain consistency throughout 👍🏼
[3a18918](https://github.com/gnolang/gno/pull/2076/commits/3a1891887cf9d5cbde51ffefb8a642149ac1b107)